organization/ircam_oauth2_provider/views.py

```diff
@@ -0,0 +1,151 @@
+            username = extra_data['username'],
+            defaults={
+                'first_name': extra_data['first_name'],
+                'last_name': extra_data['last_name'],
+                'is_active': True,
+                'is_superuser': False,
+                'is_staff': False,
+                'email': extra_data['email'],
+                'password': 'unique.password'
+            }
+        )
+        user.save()
+        self.logger.info('Local user {0} id:{1} username:{2} email:{3}'.format('created: ' if created else 'updated: ',user.id,user.username,user.email))
+        
+        # Creating or updating Person
+        self.create_or_update_person(user)
+       
+        # Creating the allauth social account so the user can log in through Ircam Auth
+        social_account = SocialAccount(user=user,
+                                    provider='ircamauth',
+                                    uid=extra_data['id'],
+                                    extra_data={'id': user.id, 'username': user.username, 'email': user.email})
+        social_account.save()
+        self.logger.info('Social account created. User {0} '.format(user.username))
+
+        return self.get_provider().sociallogin_from_response(request, extra_data)
+
+    def update_localuser(self,hello,extra_data):
+        from django.contrib.auth.models import User
+
+        user = User.objects.filter( username__icontains=extra_data['username'] )[0]
+        if (  
+                user.email != extra_data['email'] or 
+                user.first_name != extra_data['first_name'] or 
+                user.last_name != extra_data['last_name'] 
+            ) :
+            user.username = extra_data['username']
+            user.email = extra_data['email']
+            user.first_name = extra_data['first_name']
+            user.last_name = extra_data['last_name']
+            user.is_active = True
+            user.save()
+            self.logger.info('User {0} updated, email:{1}, firstN:{2}, LastN:{3}'
+                .format(user.username,user.email,user.first_name,user.last_name))
+            # Creating or updating Person
+            self.create_or_update_person(user)
+ 
+
+    def complete_login(self, request, app, token, **kwargs):
+        headers = {'Authorization': 'Bearer {0}'.format(token.token)}
+        resp = requests.get(self.profile_url, headers=headers)
+        extra_data = resp.json()
+
+        self.logger.debug("EXTRA_DATA:")
+        self.logger.debug(str(extra_data))
+        try:
+            hello = self.get_provider().sociallogin_from_response(request, extra_data)
+            saccount = SocialAccount.objects.get(
+                provider=hello.account.provider, uid=hello.account.uid
+            )
+            self.update_localuser(hello,extra_data)
+        except:
+            hello = self.create_user_socialaccount(request,extra_data)
+
+        self.logger.info('User Connexion accepted: {0} '.format(hello.user.username))
```

Comment:
Replace logs with "User logged in : {0}".
Connexion is a french word

Reply:
done